napari/_qt/menus/file_menu.py

```diff
@@ -43,6 +43,29 @@ def __init__(self, window: 'Window'):
                 'slot': window._qt_viewer._open_folder_dialog,
                 'shortcut': 'Ctrl+Shift+O',
             },
+            {
+                'menu': trans._('Open with Plugin'),
+                'items': [
+                    {
+                        'text': 'Open File(s)...',
+                        'slot': lambda: window._qt_viewer._open_files_dialog(
+                            choose_plugin=True
+                        ),
+                    },
+                    {
+                        'text': 'Open Files as Stack...',
+                        'slot': lambda: window._qt_viewer._open_files_dialog_as_stack_dialog(
+                            choose_plugin=True
+                        ),
+                    },
+                    {
+                        'text': 'Open Folder...',
+                        'slot': lambda: window._qt_viewer._open_folder_dialog(
+                            choose_plugin=True
+                        ),
+                    },
+                ],
```

Comment:
I do not like the `lambda` here. It may lead to leaking memory. 
`psygnal` handle `partial` properly but ono other of our used signal system handle it properly. And If I correctly remember, non of them handle properly `lambda`


I prefer to create a private method for this instead of `lambda`. Also this may be an inspiration for new contributors to use `lambda` in more fragile places

Reply:
Good point, will make private method above and use that. Thanks @Czaki.